File: data/skills.js
const skills = [
  {
    category: "frontend",
    title: "Frontend Engineering",
    description: "Building responsive, hypermedia-driven, and accessible user interfaces with zero bloat.",
    items: [
      { name: "HTMX & Hypermedia Architecture", level: 98, experience: "4+ yrs", icon: "fa-bolt" },
      { name: "Vanilla JavaScript (ESNext / Web Components)", level: 95, experience: "7+ yrs", icon: "fa-code" },
      { name: "Modern CSS (Grid, Flexbox, Animations, Design Tokens)", level: 96, experience: "7+ yrs", icon: "fa-palette" },
      { name: "React & Next.js Ecosystem", level: 90, experience: "5+ yrs", icon: "fa-atom" },
      { name: "TypeScript & Type Safety Systems", level: 92, experience: "5+ yrs", icon: "fa-shield-halved" },
      { name: "Performance & Lighthouse Optimization (Core Web Vitals)", level: 95, experience: "6+ yrs", icon: "fa-gauge-high" }
    ]
  },
  {
    category: "backend",
    title: "Backend & Systems",
    description: "Architecting high-throughput distributed microservices, low-latency APIs, and database engines.",
    items: [
      { name: "Node.js & Express / Fastify Frameworks", level: 96, experience: "6+ yrs", icon: "fa-server" },
      { name: "Rust (Async Tokio, Systems, Memory Management)", level: 88, experience: "3+ yrs", icon: "fa-gear" },
      { name: "Go (Distributed Services, Concurrency, Channels)", level: 90, experience: "4+ yrs", icon: "fa-cubes" },
      { name: "PostgreSQL, Redis & Distributed KV Stores", level: 94, experience: "6+ yrs", icon: "fa-database" },
      { name: "gRPC, Protocol Buffers & WebSockets", level: 91, experience: "4+ yrs", icon: "fa-network-wired" },
      { name: "Python (FastAPI, AsyncIO, Data Pipelines)", level: 89, experience: "5+ yrs", icon: "fa-brain" }
    ]
  },
  {
    category: "devops",
    title: "Cloud & Infrastructure",
    description: "Automating cloud infrastructure, CI/CD deployment pipelines, containerization, and monitoring.",
    items: [
      { name: "Docker & Container Runtime Orchestration", level: 95, experience: "6+ yrs", icon: "fa-docker" },
      { name: "Kubernetes, Helm & Cloud-Native Services", level: 88, experience: "4+ yrs", icon: "fa-dharmachakra" },
      { name: "Terraform & Infrastructure as Code (IaC)", level: 87, experience: "3+ yrs", icon: "fa-cloud" },
      { name: "GitHub Actions & Automated Testing CI/CD", level: 94, experience: "5+ yrs", icon: "fa-code-branch" },
      { name: "Prometheus, Grafana & eBPF Telemetry", level: 86, experience: "3+ yrs", icon: "fa-chart-line" },
      { name: "Linux Administration, Bash & Edge Security", level: 92, experience: "7+ yrs", icon: "fa-terminal" }
    ]
  },
  {
    category: "architecture",
    title: "Architecture & Methodology",
    description: "Designing resilient event-driven systems, RESTful hypermedia APIs, and domain-driven software.",
    items: [
      { name: "Hypermedia Driven Application Design (REST / HATEOAS)", level: 98, experience: "4+ yrs", icon: "fa-sitemap" },
      { name: "Event-Driven Architecture & Message Queues (Kafka / RabbitMQ)", level: 90, experience: "4+ yrs", icon: "fa-diagram-project" },
      { name: "Domain-Driven Design (DDD) & Clean Architecture", level: 92, experience: "5+ yrs", icon: "fa-layer-group" },
      { name: "Zero-Trust Security & OAuth2 / OIDC", level: 88, experience: "4+ yrs", icon: "fa-lock" },
      { name: "API Gateway Design & Rate Limiting", level: 93, experience: "5+ yrs", icon: "fa-filter" }
    ]
  }
];

module.exports = skills;

File: internal/data/skills.go
package data

import (
	"fmt"
	"html"
	"strings"
)

type SkillItem struct {
	Name       string `json:"name"`
	Level      int    `json:"level"`
	Experience string `json:"experience"`
	Icon       string `json:"icon"`
}

type SkillCategory struct {
	Category    string      `json:"category"`
	Title       string      `json:"title"`
	Description string      `json:"description"`
	Items       []SkillItem `json:"items"`
}

var Skills = []SkillCategory{
	{
		Category:    "frontend",
		Title:       "Frontend & Hypermedia",
		Description: "Building responsive, hypermedia-driven, and accessible user interfaces with zero client bloat.",
		Items: []SkillItem{
			{Name: "HTMX & Hypermedia Architecture", Level: 98, Experience: "4+ yrs", Icon: "fa-bolt"},
			{Name: "Vanilla JavaScript (ESNext / Web Components)", Level: 95, Experience: "7+ yrs", Icon: "fa-code"},
			{Name: "Modern CSS (Grid, Flexbox, Animations, Design Tokens)", Level: 96, Experience: "7+ yrs", Icon: "fa-palette"},
			{Name: "React & Next.js Ecosystem", Level: 90, Experience: "5+ yrs", Icon: "fa-atom"},
			{Name: "TypeScript & Type Safety Systems", Level: 92, Experience: "5+ yrs", Icon: "fa-shield-halved"},
			{Name: "Performance & Lighthouse Optimization (Core Web Vitals)", Level: 95, Experience: "6+ yrs", Icon: "fa-gauge-high"},
		},
	},
	{
		Category:    "backend",
		Title:       "Backend & Distributed Systems",
		Description: "Architecting high-throughput microservices, low-latency Go & Rust APIs, and memory engines.",
		Items: []SkillItem{
			{Name: "Go (Golang Systems, Concurrency, Channels, HTTP)", Level: 96, Experience: "5+ yrs", Icon: "fa-cubes"},
			{Name: "Node.js & Express / Fastify Frameworks", Level: 94, Experience: "6+ yrs", Icon: "fa-server"},
			{Name: "Rust (Async Tokio, Memory Safety, KV Stores)", Level: 88, Experience: "3+ yrs", Icon: "fa-gear"},
			{Name: "PostgreSQL, Redis & Distributed KV Stores", Level: 94, Experience: "6+ yrs", Icon: "fa-database"},
			{Name: "gRPC, Protocol Buffers & WebSockets", Level: 91, Experience: "4+ yrs", Icon: "fa-network-wired"},
			{Name: "Python (FastAPI, AsyncIO, Data Pipelines)", Level: 89, Experience: "5+ yrs", Icon: "fa-brain"},
		},
	},
	{
		Category:    "devops",
		Title:       "DevOps & Cloud Infrastructure",
		Description: "Automating cloud infrastructure, CI/CD deployment pipelines, containerization, and monitoring.",
		Items: []SkillItem{
			{Name: "Docker & Container Runtime Orchestration", Level: 95, Experience: "6+ yrs", Icon: "fa-docker"},
			{Name: "Kubernetes, Helm & Cloud-Native Services", Level: 88, Experience: "4+ yrs", Icon: "fa-dharmachakra"},
			{Name: "Terraform & Infrastructure as Code (IaC)", Level: 87, Experience: "3+ yrs", Icon: "fa-cloud"},
			{Name: "GitHub Actions & Automated Testing CI/CD", Level: 94, Experience: "5+ yrs", Icon: "fa-code-branch"},
			{Name: "Prometheus, Grafana & eBPF Telemetry", Level: 86, Experience: "3+ yrs", Icon: "fa-chart-line"},
			{Name: "Linux Administration, Bash & Edge Security", Level: 92, Experience: "7+ yrs", Icon: "fa-terminal"},
		},
	},
	{
		Category:    "architecture",
		Title:       "System Architecture & Methodology",
		Description: "Designing resilient event-driven systems, RESTful hypermedia APIs, and domain-driven software.",
		Items: []SkillItem{
			{Name: "Hypermedia Driven Application Design (REST / HATEOAS)", Level: 98, Experience: "4+ yrs", Icon: "fa-sitemap"},
			{Name: "Event-Driven Architecture & Message Queues (Kafka / NATS)", Level: 90, Experience: "4+ yrs", Icon: "fa-diagram-project"},
			{Name: "Domain-Driven Design (DDD) & Clean Architecture", Level: 92, Experience: "5+ yrs", Icon: "fa-layer-group"},
			{Name: "Zero-Trust Security & OAuth2 / OIDC", Level: 88, Experience: "4+ yrs", Icon: "fa-lock"},
			{Name: "API Gateway Design & Rate Limiting", Level: 93, Experience: "5+ yrs", Icon: "fa-filter"},
		},
	},
}

func RenderSkillsHTML(catParam string) string {
	targetCat := strings.ToLower(strings.TrimSpace(catParam))
	if targetCat == "" {
		targetCat = "frontend"
	}

	var group SkillCategory = Skills[0]
	for _, s := range Skills {
		if strings.ToLower(s.Category) == targetCat {
			group = s
			break
		}
	}

	var sb strings.Builder
	sb.WriteString(fmt.Sprintf(`
	<div class="skills-grid-container" id="skills-grid">
		<div class="category-info">
			<h3>%s</h3>
			<p>%s</p>
		</div>
		<div class="skill-cards">`,
		html.EscapeString(group.Title),
		html.EscapeString(group.Description),
	))

	for _, item := range group.Items {
		cardHTML := fmt.Sprintf(`
			<div class="skill-card">
				<div class="skill-card-header">
					<div class="skill-icon"><i class="fa-solid %s"></i></div>
					<div class="skill-titles">
						<h4>%s</h4>
						<span class="skill-exp">%s experience</span>
					</div>
					<span class="skill-pct">%d%%</span>
				</div>
				<div class="skill-bar-bg">
					<div class="skill-bar-fill" style="width: %d%%"></div>
				</div>
			</div>`,
			html.EscapeString(item.Icon),
			html.EscapeString(item.Name),
			html.EscapeString(item.Experience),
			item.Level,
			item.Level,
		)
		sb.WriteString(cardHTML)
	}

	sb.WriteString(`
		</div>
	</div>`)

	return sb.String()
}
